dspace-server-webapp/src/main/java/org/dspace/app/rest/submit/step/validation/CclicenseValidator.java

```diff
@@ -0,0 +1,135 @@
+     * @param configurationService DSpace configuration provided by the DI container.
+     */
+    @Inject
+    public CclicenseValidator(ConfigurationService configurationService) {
+        this.configurationService = configurationService;
+    }
+
+    private final ConfigurationService configurationService;
+
+    @Autowired
+    private ItemService itemService;
+
+    @Autowired
+    private CreativeCommonsServiceImpl creativeCommonsService;
+
+    public static final String ERROR_VALIDATION_CCLICENSEREQUIRED = "error.validation.cclicense.required";
+
+    private String name;
+
+    private Boolean required;
+
+    /**
+     * Validate the license of the item.
+     * @param item The item whose cclicense is to be validated.
+     * @param config The configuration for the submission step for cclicense.
+     * @return A list of validation errors.
+     */
+    private List<ErrorRest> validateLicense(Item item, SubmissionStepConfig config) {
+        List<ErrorRest> errors = new ArrayList<>();
+
+        String licenseURI = creativeCommonsService.getLicenseURI(item);
+        if (licenseURI == null || licenseURI.isBlank()) {
+            addError(errors, ERROR_VALIDATION_CCLICENSEREQUIRED, "/" + OPERATION_PATH_SECTIONS + "/" + config.getId());
+        }
+
+        return errors;
+    }
+
+    public ItemService getItemService() {
+        return itemService;
+    }
+
+    public void setItemService(ItemService itemService) {
+        this.itemService = itemService;
+    }
+
+    /**
+     * Check if at least one Creative Commons License is required when submitting a new Item.
+     * @return true if a Creative Commons License is required setting true for the property cc.license.required.
+     */
+    public Boolean isRequired() {
+        if (required == null) {
+            return configurationService.getBooleanProperty("cc.license.required", false);
+        }
+        return required;
+    }
+
+    /**
+     * Set whether at least one Creative Commons License is required when submitting a new Item.
+     * @param required true if a Creative Commons License is required.
+     */
+    public void setRequired(Boolean required) {
+        this.required = required;
+    }
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Perform validation on the item and config(ccLicense).
+     * @param obj The submission to be validated.
+     * @param config The configuration for the submission step for cclicense.
+     * @return A list of validation errors.
+     * @throws SQLException If there is a problem accessing the database.
+     */
+
+    @Override
+    public List<? extends ErrorRest> validate(SubmissionService submissionService,
+                                              InProgressSubmission obj,
+                                              SubmissionStepConfig config)
+            throws DCInputsReaderException, SQLException {
+        List<ErrorRest> errors = new ArrayList<>();
```

Comment:
Since we call only validateLicense to fillout the errors arraylist, you can improve this method by using an if/else statement
```java
if (this.isRequired()) {
  return validateLicense(....);
} else {
  return List.of(); // empty list
}
```